imap_processing/mag/l1a/mag_l1a.py

```diff
@@ -29,39 +34,39 @@ def mag_l1a(packet_filepath: Path, data_version: str) -> list[Path]:
 
     Returns
     -------
-    generated_files : list[pathlib.Path]
+    generated_files : list[xarray.Dataset]
         A list of generated filenames.
     """
     packets = decom_mag.decom_packets(packet_filepath)
 
     norm_data = packets["norm"]
     burst_data = packets["burst"]
 
-    generated_files = process_and_write_data(
-        norm_data,
-        mag_cdf_attrs.mag_l1a_norm_raw_attrs.output(),
-        mag_cdf_attrs.mag_l1a_norm_mago_attrs.output(),
-        mag_cdf_attrs.mag_l1a_norm_magi_attrs.output(),
-        data_version,
-    )
-    generated_files += process_and_write_data(
-        burst_data,
-        mag_cdf_attrs.mag_l1a_burst_raw_attrs.output(),
-        mag_cdf_attrs.mag_l1a_burst_mago_attrs.output(),
-        mag_cdf_attrs.mag_l1a_burst_magi_attrs.output(),
-        data_version,
+    input_files = [packet_filepath.name]
+
+    # Create attribute manager and add MAG L1A attributes and global variables
+    attribute_manager = ImapCdfAttributes()
+    attribute_manager.add_instrument_global_attrs("mag")
+    attribute_manager.add_instrument_variable_attrs("mag", "l1")
+
+    attribute_manager.add_global_attribute("Data_version", data_version)
+    attribute_manager.add_global_attribute("Input_files", str(input_files))
+    attribute_manager.add_global_attribute(
```

Comment:
Nice, this seems useful for everyone, should we add this metadata to all datasets? (Can be done in a follow-up PR if you think it'd be useful)

Reply:
Sure, I think it's worth adding to other files. Maybe we could start a document for "required SDC metadata global variables" since we have a couple